google/cloud/spanner_dbapi/exceptions.py

```diff
@@ -27,7 +29,65 @@ class Error(Exception):
     Does not include :class:`Warning`.
     """
 
-    pass
+    def _is_error_cause_instance_of_google_api_exception(self):
```

Comment:
It probably should be done in `__init__`. Cause of an error doesn't change, so it's a static property - better designate it once and keep as a `bool` attribute than check every time.

Reply:
We can't do it under the `__init__ ` method, because `__cause__` gets set after the object is created. 
so `raise EXCEPTION from CAUSE) is equivalent to 
`exc = EXCEPTION
exc.__cause__ = CAUSE
raise exc`